crates/client/src/api.rs

```diff
@@ -401,12 +401,12 @@ impl Client {
             }
         }
 
-        let map_proof_bundle: MapProofBundle<Sha256, MapLeaf> =
+        let map_proof_bundle: MapProofBundle<Sha256, LogId, MapLeaf> =
             MapProofBundle::decode(response.map.as_slice())?;
         let map_inclusions = map_proof_bundle.unbundle();
         for (leaf, proof) in leafs.iter().zip(map_inclusions.iter()) {
             let found = proof.evaluate(
-                &leaf.log_id,
+                leaf.log_id.clone(),
```

Comment:
If `ReversePath` is made to take `&K` in its constructor, I think we can back out this change.